.travis.yml

```diff
@@ -39,19 +43,20 @@ before_install:
   - echo $ANDROID_HOME
   - |
     if [ ${START_EMU} = "1" ]; then
-      cat /proc/sys/net/ipv6/conf/all/disable_ipv6
-      cat /etc/sysctl.conf
+      echo 'count=0' > /home/travis/.android/repositories.cfg
       echo y | android update sdk --no-ui -t tools
-      echo yes | sdkmanager tools > /dev/null
+      echo y | sdkmanager "platform-tools" >/dev/null
+      echo y | sdkmanager tools > /dev/null
       # Free up some space
-      for apiVersion in 15 16 17 18 19; do
-        sdkmanager --uninstall "system-images;android-${apiVersion};default;armeabi-v7a" > /dev/null
+      for apiVersion in 18 19; do
         sdkmanager --uninstall "platforms;android-${apiVersion}" > /dev/null
       done
-      echo yes | sdkmanager --update > /dev/null
-      echo yes | sdkmanager "platforms;${ANDROID_EMU_TARGET}" > /dev/null
-      echo yes | sdkmanager "extras;android;m2repository" > /dev/null
-      # echo yes | sdkmanager "extras;google;m2repository" > /dev/null
+      echo y | sdkmanager "build-tools;28.0.3" >/dev/null
+      echo y | sdkmanager --update > /dev/null
+      echo y | sdkmanager "platforms;${ANDROID_EMU_TARGET}" > /dev/null
+      echo y | sdkmanager "platforms;android-28" >/dev/null
+      echo y | sdkmanager "extras;android;m2repository" > /dev/null
+      echo y | sdkmanager --channel=4 "emulator" >/dev/null
```

Comment:
what is channel4?

https://developer.android.com/studio/command-line/sdkmanager says it should be in range 0-3

Reply:
Specifies Canary.

Reply:
> --channel=channel_id | Include packages in channels up to channel_id. Available channels are:0 (Stable), 1 (Beta), 2(Dev), and 3 (Canary).

Perhaps my tutorial is different than yours %)